application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
```

Comment:
Feel free to push right away. (Just don't amend and force-push to ease reviewability.) We can continue to discuss on a latest commit (and should be easier too).